### PR TITLE
Add Legion Go Initial Controller Support

### PR DESCRIPTION
**Pull request by aarron-lee**
# Details

This PR adds initial support for the Lenovo Legion Go's controller while in X-input mode.

It does the following:

- `Legion_L + X` maps to the onscreen keyboard
- `Legion_L + A` maps to the QAM
- `Legion_L + B` maps to the Steam/Guide button

Note that it also now seems to suspend properly when pressing the power button, tapping the power button now shows the Steam suspend animation while in gamescope-session.

Tested on the Legion Go with the controller in X-input mode while both attached and detached, confirmed working in both cases.

Edit: noticed that some users can't use the Legion + B combination, we suspect a controller firmware update is the reason why.
Those that did the firmware update can no longer see the Legion + B combo register anything with evtest

# Device Info:

tested on:

Lenovo Legion Go
ChimeraOS 45 (unstable)
kernel 6.6.0-chos1-chimeraos-2
BIOS N3CN22WW

### CODE REVIEW

File: src/handycon/handhelds/go_gen1.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    # Button map shortcuts for easy reference.
+    button2 = handycon.button_map["button2"]  # Default QAM
+    button4 = handycon.button_map["button4"]  # Default OSK
+    button5 = handycon.button_map["button5"]  # Default MODE
+
+    ## Loop variables
+    button_on = seed_event.value
+
+    # Legion + x = keyboard
+    if active_keys == [99] and button_on == 1 and button4 not in handycon.event_queue:
```
Comment (pastaq) [inline src/handycon/handhelds/go_gen1.py:37]: Minor nitpicking, but can you put the buttons in numerical order? This event trigger should follow button 2.

File: src/handycon/utilities.py [hunk 2]
```diff
@@ -175,6 +176,12 @@ def id_system():
         handycon.system_type = "AYN_GEN3"
         ayn_gen3.init_handheld(handycon)
```
Comment (pastaq) [inline src/handycon/utilities.py:178]: Can you add a Lenovo section header to the comments in the same style as the other manufacturers please.

**Review by pastaq — CHANGES_REQUESTED** [review]
Looks good, just a couple formatting things need changed.

File: src/handycon/handhelds/go_gen1.py (hunk 1 above)
Comment (aarron-lee) [reply src/handycon/handhelds/go_gen1.py]: done

File: src/handycon/utilities.py (hunk 2 above)
Comment (aarron-lee) [reply src/handycon/utilities.py:178]: done

**Pull request merged**